activerecord/test/cases/adapters/postgresql/postgresql_adapter_test.rb

```diff
@@ -649,7 +649,25 @@ def test_does_not_raise_notice_level_warnings
         end
       end
 
+      def test_date_decoder
```

Comment:
Could you add another test that shows behaviour when the config is off (the default)?